### ISSUE TITLE
ROB-SYSID-001: robotics system identification with identifiability and held-out validation

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#4801]
Parent: #4795. Depends on ROB-EXP-001 #4800 and ROB-DESIGN parameter provenance #4797. Coordinate with #673 sim-to-real transfer, #664 causal-mechanism qualification, #1430 controller-adaptation quarantine, and #1738 ETK.

## Purpose

Add a robotics-specific system-identification layer that estimates bounded plant/model parameters from exact physical experiment evidence while preserving parameter ambiguity, dependence, applicability, and held-out validation.

The output is a **model-update proposal / identified-parameter evidence artifact**, never direct mutation of a live qualified model or controller.

## Core theorem

```text
best fit
!= uniquely identified parameters
!= causal mechanism truth
!= qualified plant model
!= controller authority
```

and:

```text
training fit != held-out validation
```

## Inputs

A V1 identification campaign should bind:

- exact `ExperimentRunId` set from #4800;
- exact test article/design/as-built/topology/calibration identities;
- exact parameter/model family from #4797/#4799;
- exact free vs fixed parameter set;
- exact units/conventions;
- exact preprocessing/filtering/downsampling identity;
- exact loss/objective definition;
- exact optimizer/solver implementation/version;
- initial/prior parameter bounds;
- random seeds where material;
- fit/validation split identity;
- resource/termination criteria.

## Output

Conceptually:

```rust
pub struct SystemIdentificationResultV1 {
    pub campaign_id: SysIdCampaignId,
    pub model_subject: ModelSubjectRef,
    pub fit_runs: Vec<ExperimentRunId>,
    pub validation_runs: Vec<ExperimentRunId>,
    pub parameter_estimates: Vec<IdentifiedParameterV1>,
    pub identifiability: IdentifiabilityAssessmentV1,
    pub residuals: ResidualAnalysisV1,
    pub validation: HeldOutValidationV1,
    pub applicability: IdentifiedModelApplicabilityV1,
    pub disposition: IdentificationDispositionV1,
}
```

Exact names can change.

## Identifiability is first-class

V1 must represent at least:

```text
IdentifiedWithinProfile
WeaklyIdentified
PartiallyIdentified
NonIdentifiable
OutOfExperimentDomain
```

A precise optimizer result cannot upgrade `NonIdentifiable` to identified.

Assess identifiability using one or more explicit methods appropriate to the model/profile, for example:

- sensitivity/Jacobian rank or conditioning;
- profile likelihood / parameter sweeps;
- posterior/covariance structure where justified;
- multi-start solution degeneracy;
- designed synthetic canaries with known ambiguity.

The exact method identity and limitations must be part of evidence.

## Parameter correlation / equivalence classes

When multiple parameter combinations explain the same data, preserve the relationship rather than selecting one arbitrary point.

Examples:

- inertia ↔ damping tradeoff;
- torque constant ↔ transmission efficiency;
- Coulomb friction ↔ unmodeled deadband;
- compliance ↔ sensor/actuator latency in limited experiments.

Support parameter dependence groups, covariance/bounds, or explicit alternate-solution sets rather than pretending independent certainty.

## Held-out validation

Require a distinct held-out experiment set for positive predictive-identification claims.

```text
fit data
!= validation data
```

The same run may not receive independent validation credit after influencing optimizer selection, bounds, preprocessing choice, model-family choice, or hyperparameter tuning unless an explicit nested-validation protocol accounts for that dependence.

Validation should preserve metrics by observable rather than one scalar fit score.

Potential metrics:

- position trajectory error;
- velocity error;
- current error;
- torque/load error;
- temperature error;
- phase/latency error;
- steady-state bias;
- transient overshoot/settling mismatch.

## Residual diagnostics

Preserve structured residual evidence, including when useful:

- time-correlated residuals;
- load/speed/temperature dependence;
- saturation regions;
- systematic sign/asymmetry;
- heteroscedasticity;
- unmodeled periodicity/cogging;
- regime-dependent failure.

A persistent structured residual should be able to produce `ModelInadequacySuspected`, not merely higher optimizer loss.

## Model-family selection

Do not force every mismatch into parameter values.

Compare candidate model families where appropriate, e.g.:

```text
viscous-only
vs Coulomb+viscous
vs Stribeck/LuGre-style
vs explicit backlash/compliance
```

A more complex family must not automatically win because it fits training data better. Preserve complexity/model-selection criteria and held-out evidence.

## Experiment insufficiency / next-experiment proposal

If parameters are weak/non-identifiable, emit a bounded **discriminating-experiment proposal**, for example:

- no-load speed sweep;
- bidirectional slow sweep;
- load step;
- chirp/frequency sweep;
- thermal profile;
- alternate payload.

This proposal has no actuation authority. ROB-DOE may later optimize these proposals under risk/wear/cost constraints.

## Update boundary

Successful identification creates a new inert candidate parameter/model revision:

```text
current model M0
+ identification evidence
-> candidate M1
```

It must not mutate M0 in place when M0 participates in qualified simulation, safety, or controller semantics.

Promotion to a current engineering/plant model must pass the applicable ETK/change-impact/qualification path.

For controller-related adaptations, #1430 remains authoritative.

## Causal/mechanism claim boundary

A model may predict held-out trajectories well while its internal physical parameters are wrong.

Therefore:

```text
predictive system identification
!= causal mechanism qualification
```

Use #664 for stronger causal-mechanism claims when needed.

## Required synthetic qualification fixtures

Before physical JointLab evidence, include known-answer synthetic fixtures such as:

1. identifiable single-inertia + damping system;
2. intentionally under-excited system with non-identifiable parameters;
3. correlated parameter pair producing near-equivalent outputs;
4. wrong model family producing structured residuals;
5. training overfit with held-out failure;
6. correct model with measurement noise;
7. latency mismatch mistaken for damping without timing evidence;
8. parameter outside declared bounds;
9. stale calibration/test article mismatch;
10. duplicated/replayed experiment runs not counted as independent evidence.

## Required hostile tests

- fit and validation set overlap is detected according to protocol;
- same experiment copied under another filename/ID cannot create independent evidence;
- wrong physical article cannot join the campaign through friendly name;
- stale calibration fails when required;
- unit mismatch rejects;
- optimizer success with non-identifiable Jacobian remains non/weakly identified;
- optimizer failure remains distinct from negative physical result;
- parameter at bound is surfaced;
- non-finite residual/estimate rejects;
- model-family substitution changes campaign/result identity;
- preprocessing change changes identity;
- changing one fit sample/result bundle changes result identity;
- candidate identified model cannot deserialize into live qualified controller/model authority.

## Nonclaims

ROB-SYSID-001 does not establish physical-law discovery, causal truth, controller safety, requirement satisfaction, production calibration, or actuation authority.

## PR shape

Start with a pure analysis/evidence contract and deterministic synthetic known-answer harness. Physical JointLab campaigns should be a later evidence tranche once ROB-EXP and commissioning/calibration prerequisites are actually executable-qualified.

**Comment by Tristan-Stoltz-ERC** [comment]
Numerical substrate audit refinement:

ROB-SYSID should now treat #4867 (`STAT-LM-001`) as the preferred shared prerequisite for **linear known-answer identification** rather than depending on archived `symthaea-statistics-v*` migration docs.

Live repository audit shows:

```text
current symthaea-statistics
  -> simple univariate OLS (`linear_regression`)
  -> descriptive / inference / Bayesian helpers

archived migration/version artifacts
  -> mention QR least-squares, ridge, diagnostics, local-level Kalman
  -> NOT the current live exported API
```

`symthaea-control` already provides control/state-estimation primitives such as a linear Kalman filter, but that is also a different proposition from parameter identification.

Recommended decomposition:

```text
#4867 STAT-LM
  checked multivariate numerical regression
  rank/conditioning disposition
  residual/covariance diagnostics
        ↓
#4801 ROB-SYSID
  ExperimentRun lineage
  physical model/parameter family
  fit vs holdout split
  identifiability interpretation
  applicability
  model discrepancy
  candidate model-update proposal
```

Freeze:

```text
QR full-column-rank
!= physical/causal identifiability

low training residual
!= held-out model validity
```

A useful first ROB-SYSID synthetic fixture after #4867 qualifies remains a single-joint linearized plant with known inertia/damping, followed immediately by an intentionally under-excited rank-deficient fixture.

**Comment by Tristan-Stoltz-ERC** [comment]
### 2026-09-20 audit — separate numerical rank from excitation sufficiency

STAT-LM-001A / #4868 introduces a checked numerical least-squares substrate with explicit rank/conditioning evidence. ROB-SYSID should **not** treat `FullColumnRank` from that solver as sufficient system-identification excitation evidence.

Freeze the stronger boundary:

```text
full-column-rank sampled design matrix
!= persistently/informatively excited physical experiment
!= parameter sensitivity over the declared operating envelope
!= robust identifiability under noise / timing / saturation
```

For each identification campaign, add an explicit `ExcitationAssessmentV1` (name can change) or equivalent evidence carrying at least:

- exact input/excitation protocol/run identities;
- parameter/model family being tested;
- sensitivity/regressor construction identity;
- singular values / information-matrix spectrum or equivalent method output;
- declared rank/conditioning threshold profile;
- frequency/amplitude/operating-region coverage where relevant;
- command and sensor saturation/dropout coverage;
- duration / sample-count / independent-condition evidence;
- disposition such as `SufficientWithinProfile`, `WeakExcitation`, `MissingRegime`, `Saturated`, `TimingIndeterminate`, or `InsufficientEvidence`.

A regression can be algebraically full-rank yet practically useless because the smallest singular direction is dominated by measurement noise, one parameter is excited only in a tiny regime, or the relevant dynamic frequencies were never stimulated.

Synthetic fixtures should therefore include at least:

1. algebraically full-rank but extremely weak excitation;
2. position-rich but acceleration-poor trajectory that cannot robustly estimate inertia;
3. velocity-rich low-frequency sweep that identifies viscous damping but not high-frequency dynamics;
4. one-direction motion that cannot discriminate asymmetric friction/backlash;
5. input excitation whose informative region is clipped by actuator saturation;
6. nominally informative chirp corrupted by timestamp skew/latency uncertainty;
7. same trajectory under two payloads improving inertia/load separation;
8. repeated samples at one operating point proving precision but not domain coverage.

ROB-SYSID may consume qualified STAT-LM numerical results, but the positive identification disposition should require the **system-identification excitation theorem** as a separate proposition.

**Comment by Tristan-Stoltz-ERC** [comment]
### 2026-09-20 audit — prevent observer/model circularity in system identification

ROB-SYSID should explicitly detect when a derived state used as an identification regressor/target was produced by a model that already contains the parameter or mechanism being estimated.

Core boundary:

```text
observer-estimated state fits candidate model
!= raw physical evidence supports candidate model
```

Examples of dangerous circularity:

- acceleration estimated by an observer using assumed inertia, then used to estimate inertia;
- torque estimated from current using an assumed torque constant/gear efficiency, then used to validate that same torque constant/efficiency;
- Kalman velocity generated from a transition model with assumed damping, then used to identify damping;
- filtered/smoothed validation data whose tuning parameters were selected to maximize candidate-model agreement.

Add an `AnalysisDependencyGraph` / transform-lineage check (name can change) capable of determining whether each fitted/validation observable is:

- direct calibrated measurement;
- model-independent deterministic derivation;
- model-dependent estimate with exact model/parameter dependencies;
- derived using fit data;
- derived/tuned using validation data.

Suggested dispositions:

```text
IndependentEnoughForProfile
ModelDependentButAdmitted
CircularDependencyDetected
ValidationLeakageDetected
DependencyIndeterminate
```

A circular dependency does not necessarily make the data useless; it blocks the stronger independent-identification claim unless an explicit joint-estimation/statistical protocol accounts for it.

Required synthetic canaries:

1. direct encoder position + finite-difference velocity remains model-independent but derivative/noise-limited;
2. Kalman velocity using the true candidate damping parameter is detected as model-dependent;
3. current→torque conversion using the same torque constant under estimation is detected;
4. independent load-cell torque remains an external check;
5. filter hyperparameters tuned on fit data are allowed only under the declared training pipeline;
6. filter/model tuning on held-out validation invalidates independent validation credit;
7. model-dependent preprocessing identity change changes the campaign/result identity.

This should compose with ROB-EXP's derived-signal provenance rather than banning state estimation. The goal is to make **which assumptions entered the observations** inspectable before claiming parameter identification.
